tests/sparseml/pytorch/image_classification/test_export.py

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+
+import onnx
+import pytest
+import torch
+
+from click.testing import CliRunner
+from sparseml.pytorch.image_classification.export import main
+from sparseml.pytorch.models import resnet50
+from sparsezoo.analysis import ModelAnalysis
+
+
+@pytest.fixture()
+def temp_dir(tmp_path):
+    """
+    return: a temporary directory
+    """
+    test_dir = tmp_path / "test_dir"
+    test_dir.mkdir()
+    yield test_dir
+
+
+@pytest.fixture()
+def non_quant_resnet_checkpoint(temp_dir):
+    """
+    return: path to a ResNet50 checkpoint
+    """
+    checkpoint_path = str(temp_dir / "model.pth")
+    model = resnet50()
+    torch.save(model, checkpoint_path)
+    yield checkpoint_path
+
+
+@pytest.fixture()
+def quant_recipe_path():
+    """
+    return: path to a resnet50 recipe with quantization
+    """
+    yield str(Path(__file__).resolve().parent / "resnet_test_recipe.yaml")
+
+
+def test_export_one_shot(non_quant_resnet_checkpoint, quant_recipe_path, temp_dir):
+    runner = CliRunner()
+    arch_key = "resnet50"
+    result = runner.invoke(
+        main,
+        [
+            "--arch-key",
+            "resnet50",
+            "--checkpoint-path",
+            non_quant_resnet_checkpoint,
+            "--num-samples",
+            0,
+            "--save-dir",
+            temp_dir,
+            "--num-classes",
+            1000,
+            "--one-shot",
+            quant_recipe_path,
+        ],
+    )
+    assert result.exit_code == 0
+
+    expected_onnx_path = str(temp_dir / arch_key / "deployment" / "model.onnx")
+
+    # exported file exists
+    assert Path(expected_onnx_path).exists()
+
+    # exported file is valid onnx
+    model = onnx.load(expected_onnx_path)
+    onnx.checker.check_model(model)
+
+    # check onnx model is quantized
+    assert any(node.op_type == "QuantizeLinear" for node in model.graph.node)
+
+    # check onnx model is sparse
+    model_analysis = ModelAnalysis.from_onnx(onnx_file_path=expected_onnx_path)
+    block = model_analysis.operation_summary.ops.block_structure
+
+    assert block["single"].sparsity > 0.90
```

Comment:
this still doesn't quite add up (doesn't check the previous issue that the model was way too sparse, likely all 0s) - sparsity shouldn't be greater than 0.9 since we are targeting 0.9. should test that we are within a margin of error.

Reply:
What's an acceptable margin of error?